core/input/input.cpp

```diff
@@ -813,29 +818,27 @@ Point2i Input::warp_mouse_motion(const Ref<InputEventMouseMotion> &p_motion, con
 }
 
 void Input::action_press(const StringName &p_action, float p_strength) {
-	Action action;
+	// Create or retrieve existing action.
+	Action &action = action_state[p_action];
```

Comment:
Note this is assuming that the usage here is single threaded, and the previous deferring the setting of the action wasn't some attempt at thread protection. If the `action_state` was meant to be thread safe it should probably be done properly with e.g. mutex.

Reply:
I think it's fine that `Input` is thread-unsafe (i.e., only OK from the main thread) in general (besides `parse_input_event()`. At some point the thread guards should be uncoupled from `Node` so most singletons methods' can be decorated as only-safe-from-main and users get to know if they are doing something risky.